File: backend/app/models/departure.py
from pydantic import BaseModel
from typing import Optional

class Departure(BaseModel):
    id: str
    thumbnail: str
    description: str
    platform: str
    departure: str          # stored as HH:MM 24h

class DepartureCreate(BaseModel):
    thumbnail: str = "🚆"
    description: str
    platform: str = ""
    departure: str

class DepartureUpdate(BaseModel):
    thumbnail: Optional[str] = None
    description: Optional[str] = None
    platform: Optional[str] = None
    departure: Optional[str] = None

File: frontend/src/types/departure.ts
export interface Departure {
  id: string;
  thumbnail: string;
  description: string;
  platform: string;
  departure: string; // "HH:MM" 24h
}

export type DepartureCreate = Omit<Departure, "id">;
export type DepartureUpdate = Partial<DepartureCreate>;

export interface AuthStatus {
  authenticated: boolean;
  user: string | null;
}
